quality_control_claim/models/qc_inspection.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, fields, api, exceptions, _
+
+
+class QcInspection(models.Model):
+    _inherit = 'qc.inspection'
+
+    @api.one
+    def _count_claims(self):
+        claim_obj = self.env['crm.claim']
+        cond = [('ref', '=', '%s,%s' % (self._model, self.id))]
+        claims = claim_obj.search(cond)
+        self.claims = len(claims)
+
+    automatic_claims = fields.Boolean(
+        'Automatic Claims', default=False,
+        help="If you want to create one claim when the quality test status is"
+             " 'Quality failed'.")
+    automatic_claims_by_line = fields.Boolean(
+        'Automatic Claims by line', default=False,
+        help="If you want to create one claim per quality test line, when the"
+             " quality test line status is 'Tolerable' or 'Not Tolerable'.")
+    claims = fields.Integer(string="Created claims",
+                            compute='_count_claims', store=False)
+
+    @api.multi
+    def _prepare_inspection_header(self, object_ref, test):
+        result = super(QcInspection, self)._prepare_inspection_header(
+            object_ref, test)
+        result.update({'automatic_claims': test.automatic_claims,
+                       'automatic_claims_by_line':
+                       test.automatic_claims_by_line})
+        return result
+
+    @api.multi
+    def action_approve(self):
+        crm_claim_obj = self.env['crm.claim']
+        category_obj = self.env['crm.case.categ']
+        nc_category_search = [('name', '=', 'NC')]
+        idi_category_search = [('name', '=', 'IDI')]
+        nc_categ = category_obj.search(nc_category_search, limit=1)
+        if not nc_categ:
+            raise exceptions.Warning(
+                _("NC crm case category NOT FOUND"))
+        idi_categ = category_obj.search(idi_category_search, limit=1)
+        if not idi_categ:
+            raise exceptions.Warning(
+                _("IDI crm case category NOT FOUND"))
+        super(QcInspection, self).action_approve()
+        for inspection in self:
+            if inspection.state == 'failed' and inspection.automatic_claims:
+                vals = inspection.init_claim_vals()
+                vals['name'] = _('Quality test %s for product %s'
+                                 ' unsurpassed') % (self.name,
+                                                    self.object_id.name)
+                crm_claim_obj.create(vals)
+            if inspection.automatic_claims_by_line:
+                for line in inspection.inspection_lines:
+                    if line.tolerance_status == 'not_tolerable':
```

Comment:
Hola, buenos días a los 2, esto es lo que se me pidió en un principio.
1.- Un módulo que metiera un check "Reclamaciones Automáticas" en los test, si dicho test tiene este campo clicado, y el estado del test es erróneo, se crea una reclamación.
2.- Un modulo que crease una reclamación por cada línea de test de calidad, si el color de la línea de test es "rojo", se crea una reclamación en la categoría NC. Si el color de la línea de test es "Naranja", que Ana solicitó el amarillo, pero se cambió por el naranja, se crea una reclamación en la categoría IDI. Este tema de los colores se encuentra en el campo "tolerance_status", entonces no entiendo porque decís que esto va por el estado de la línea, y no por el campo "tolerance_status".
3.- Un módulo que cuando se crea una reclamación en categoría NC, se crea una acción correctiva.

Esto es lo que se me pidió en su día, y se me dijo que el nombre del primer módulo podría ser "quality_claim_link". Hice el primer módulo, luego el segundo módulo "quality_line_claim_link", y Pedro me dijo que los nombre no eran correctos, que le llamaría "quality_control_claim", y que metiera los puntos 1 y 2 en dicho único modulo "quality_control_claim".
Me quedo a la espera a que me digáis al final como queréis que deje las cosas.
Un saludo